frame/treasury/src/lib.rs

```diff
@@ -299,39 +357,126 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// The needed bond for a proposal whose spend is `value`.
-	fn calculate_bond(value: RingBalance<T>) -> RingBalance<T> {
-		T::ProposalBondMinimum::get().max(T::ProposalBond::get() * value)
+	fn calculate_bond(value: StakingBalanceT<T>) -> StakingBalanceT<T> {
+		match value {
+			StakingBalance::KtonBalance(value) => {
+				StakingBalance::kton(T::KtonProposalBondMinimum::get().max(T::ProposalBond::get() * value))
+			}
+			StakingBalance::RingBalance(value) => {
+				StakingBalance::ring(T::RingProposalBondMinimum::get().max(T::ProposalBond::get() * value))
+			}
+		}
 	}
 
-	// Spend some money!
-	fn spend_funds() {
-		let mut budget_remaining = Self::pot();
-		Self::deposit_event(RawEvent::Spending(budget_remaining));
+	/// TODO: Need a pre-logic to infer what specific to do with kton or ring.
```

Comment:
The budgets are from different pots, and require checking.

It is possible to reuse the code and merge spend_kton_funds and spend_ring_funds